src/app/buildings/components/building.component.ts

```diff
@@ -0,0 +1,73 @@
+import { rxResource } from '@angular/core/rxjs-interop';
+import { ChangeDetectionStrategy, Component, computed, effect, inject, input, output, Signal } from '@angular/core';
+import { BuildingService } from '../services/building.service';
+import { DEFAULT_CENTER, DEFAULT_MAP_CONFIG } from '../../maps/utils/constants';
+import { MapMarker } from '../../maps/interfaces/map-marker.interface';
+import { MapConfig } from '../../maps/interfaces/map-config.interface';
+import { of } from 'rxjs';
+import { BuildingDetails } from '../interfaces/building-details.interface';
+
+
+@Component({
+  selector: 'app-building-component',
+  imports: [],
+  template: ``,
+  styleUrl: './building.component.css',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class BuildingComponent {
+  buildingService = inject(BuildingService);
```

Comment:
Esta variable debería ser privada ya que únicamente la estamos usando dentro del .ts

```suggestion
  private _buildingService = inject(BuildingService);
```